### DISCUSSION TITLE
MLP block in AutoEncoder

### DISCUSSION THREAD

**Discussion by thibaultdvx** [Project-MONAI/MONAI#8077] [Q&A]
Hi,
I would like to know if it is possible to have an intermediate MLP block inside an AutoEncoder (from `monai.networks.nets`). It seems to me that it is only possible to have convolution layers inside this network.

Thanks

**Comment by KumoLiu** [comment]
Hi @thibaultdvx, thanks for your interest here.

 the AutoEncoder networks are typically built using convolutional layers since they are designed primarily for processing medical images, where convolutional layers excel at capturing spatial hierarchies and local patterns. However, it's certainly possible to insert an intermediate MLP block into an AutoEncoder. This would involve customizing the architecture by adding a fully connected (dense) layer after flattening the feature maps before feeding them into the bottleneck or decoder part of the network.
You can define a custom AutoEncoder class. This allows you to insert an MLP block in the latent space or any part of the network.
In MONAI, perhaps you need to modify the intermediate layer.

Hope it helps, thanks.

**Reply by thibaultdvx** [reply]
Thanks for your reply. So in conclusion, you don't plan to add arguments to create an intermediate MLP block in `monai.networks.nets.AutoEncoder`, do you?

**Reply by karllandheer** [reply]
Hello, I second this request. It is common to desire a MLP block in the case where you're more interested in the embedding than anything. Is there a tutorial on how to define a custom MONAI class?
